### PR TITLE
rendertarget_gl: fix depth stencil texture attachment on OpenGLES 2.0

### PR DESCRIPTION
**Pull request by mbouron-gpsw**
GL_OES_packed_depth_stencil does not define a single combined attachment
point (GL_DEPTH_STENCIL_ATTACHMENT), thus, we need to attach the texture
to both GL_DEPTH_ATTACHMENT and GL_STENCIL_ATTACHMENT.

### CODE REVIEW

File: tests/texture.py [hunk 1]
```diff
@@ -31,12 +31,21 @@
 from pynodegl_utils.toolbox.colors import get_random_color_buffer
 
 
+_RENDER_BUFFER_FRAG = '''
+void main()
+{
+    float color = ngl_texvideo(tex0, var_tex0_coord).r;
```
Comment (cboesch-gpsw) [inline tests/texture.py:37]: you can use `ngl_tex2d` here since we know it's a data buffer.

Comment (mbouron-gpsw) [reply tests/texture.py:37]: Done.

File: libnodegl/api.c [hunk 2]
```diff
@@ -409,6 +409,7 @@ static const char *get_cap_string_id(unsigned cap_id)
     case NGL_CAP_NPOT_TEXTURE:                  return "npot_texture";
     case NGL_CAP_TEXTURE_3D:                    return "texture_3d";
     case NGL_CAP_TEXTURE_CUBE:                  return "texture_cube";
+    case NGL_CAP_SHADER_TEXTURE_LOD:            return "shader_texture_lod";
```
Comment (cboesch-gpsw) [inline libnodegl/api.c:412]: alphabetical order please

File: libnodegl/backends/gl/glfeatures_data.h [hunk 3]
```diff
@@ -280,5 +280,12 @@ static const struct glfeature {
         .flag           = NGLI_FEATURE_SHADING_LANGUAGE_420PACK,
         .version        = 420,
         .extensions     = (const char*[]){"GL_ARB_shading_language_420pack", NULL},
+    }, {
+        .name           = "shader_texture_lod",
+        .flag           = NGLI_FEATURE_SHADER_TEXTURE_LOD,
+        .version        = 300,
+        .es_version     = 300,
+        .es_extensions  = (const char*[]){"GL_EXT_shader_texture_lod", NULL},
+
```
Comment (cboesch-gpsw) [inline libnodegl/backends/gl/glfeatures_data.h:289]: extra line break

**Review by cboesch-gpsw — APPROVED** [review]
Aside from the cosmetics, LGTM

File: libnodegl/pgcraft.c [hunk 4]
```diff
@@ -532,6 +532,7 @@ static void set_glsl_header(struct pgcraft *s, struct bstr *b, const struct pgcr
     const int require_image_external_essl3_feature = ngli_darray_count(&s->texture_infos) > 0 && s->glsl_version >= 300;
 #endif
     const int enable_shader_texture_lod = (gctx->features & NGLI_FEATURE_SHADER_TEXTURE_LOD) == NGLI_FEATURE_SHADER_TEXTURE_LOD;
+    const int enable_texture_3d = (gctx->features & NGLI_FEATURE_TEXTURE_3D) == NGLI_FEATURE_TEXTURE_3D;
```
Comment (cboesch-gpsw) [inline libnodegl/pgcraft.c:535]: vertical align?

File: libnodegl/api.c (hunk 2 above)
Comment (mbouron-gpsw) [reply libnodegl/api.c]: Done.

File: libnodegl/backends/gl/glfeatures_data.h (hunk 3 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/gl/glfeatures_data.h:290]: Fixed.

File: libnodegl/pgcraft.c (hunk 4 above)
Comment (mbouron-gpsw) [reply libnodegl/pgcraft.c]: Done.

File: libnodegl/pgcraft.c [hunk 5]
```diff
@@ -568,6 +570,15 @@ static void set_glsl_header(struct pgcraft *s, struct bstr *b, const struct pgcr
             ngli_bstr_print(b, "#define ngl_tex2d   texture2D\n"
                                "#define ngl_tex3d   texture3D\n"
                                "#define ngl_texcube textureCube\n");
+
+        if (config->backend == NGL_BACKEND_OPENGLES && s->glsl_version < 300)
+            ngli_bstr_print(b, "#define ngl_tex2dlod   texture2DLodEXT\n"
+                               "#define ngl_tex3dlod   texture3DLodEXT\n"
+                               "#define ngl_texcubelod textureCubeLodEXT\n");
```
Comment (cboesch-gpsw) [inline libnodegl/pgcraft.c:579]: Is the `EXT` symbol always available? No scenario where only `texture2DLod` would?

File: libnodegl/api.c [hunk 6]
```diff
@@ -425,6 +426,7 @@ static int load_caps(struct ngl_backend *backend, const struct gctx *gctx)
     const int has_npot_texture   = ALL_FEATURES(gctx->features, NGLI_FEATURE_TEXTURE_NPOT);
     const int has_texture_3d     = ALL_FEATURES(gctx->features, NGLI_FEATURE_TEXTURE_3D);
     const int has_texture_cube   = ALL_FEATURES(gctx->features, NGLI_FEATURE_TEXTURE_CUBE_MAP);
+    const int has_shader_texture_lod = ALL_FEATURES(gctx->features, NGLI_FEATURE_SHADER_TEXTURE_LOD);
```
Comment (cboesch-gpsw) [inline libnodegl/api.c:430]: can you keep this one sorted as well?

**Review by cboesch-gpsw — COMMENTED** [review]
> internal: sort alphabetically capabilities array

"internal: sort capabilities array alphabetically" is more correct I believe.

File: libnodegl/pgcraft.c (hunk 5 above)
Comment (mbouron-gpsw) [reply libnodegl/pgcraft.c:579]: They are only available when GL_EXT_shader_texture_lod is available and enabled.
On OpenGLES 2.0: only texture2DLodEXT is available with an extension
On OpenGLES 3.0: textureLod is always available
On OpenGL: texture2DLod is deprecated (and replaced by textureLod) in GLSL version 130 and we don't support OpenGL 2.x.

File: libnodegl/api.c (hunk 6 above)
Comment (mbouron-gpsw) [reply libnodegl/api.c]: Done.

File: libnodegl/pgcraft.c (hunk 5 above)
Comment (cboesch-gpsw) [reply libnodegl/pgcraft.c:579]: So there was never a period with `texture2DLod` in OpenGLES 2, interesting. Thanks.

**Pull request merged**